app/services/search_service.rb

```diff
@@ -0,0 +1,46 @@
+class SearchService
+  # Fetch all of the offenders (for a given prison) filtering
+  # out offenders based on the provided text.
+  # rubocop:disable Metrics/MethodLength
+  def self.search_for_offenders(text, prison)
+    number_of_requests = max_requests_count(prison)
+    tier_map = CaseInformationService.get_case_information(prison)
+
+    search_term = text.upcase
+    search_results = []
+
+    (0..number_of_requests).each do |request_no|
+      offenders = OffenderService.get_offenders_for_prison(
+        prison,
+        page_number: request_no,
+        page_size: 250,
+        tier_map: tier_map
+      )
+      break if offenders.blank?
+
+      new_offenders = offenders.select { |offender|
+        offender.last_name.start_with?(search_term) ||
```

Comment:
Does the search term need to be an exact match for the name?

Reply:
Yeah, I think they need to know the start of either the first/last name to be able to find them.

Reply:
Does Jay and the design crew know that this is not a fuzzy search?

Reply:
Yes, we also had a conversation about relevance, "disappointed but resigned" is the best description of the response.